x-pack/test/detection_engine_api_integration/utils.ts

```diff
@@ -1202,3 +1213,16 @@ export const deleteMigrations = async ({
     )
   );
 };
+
+export const getOpenSignals = async (
```

Comment:
The addition of this function implies that our `refresh` params are not effective. Is this true? It's my understanding we set refresh to `wait_for` when a rule executes, which implies we are receiving a response of successfully indexed signals before the index is refreshed. Which shouldn't happen, correct?

https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html#docs-refresh

Reply:
Interesting we only set `refresh: wait_for` if there are actions present on the rule..

https://github.com/elastic/kibana/blob/master/x-pack/plugins/security_solution/server/lib/detection_engine/signals/signal_rule_alert_type.ts#L129

Reply:
We only set `refresh = 'wait_for'` if there are actions on the rule. In local testing I was getting false positive test results while writing tests for exceptions, i.e. the rule generated signals but querying the signals index returned no signals. In 15 test runs after adding the explicit refresh I didn't see any flakiness.

Reply:
> In local testing I was getting false positive test results while writing tests for exceptions, i.e. the rule generated signals but querying the signals index returned no signals. In 15 test runs after adding the explicit refresh I didn't see any flakiness.

Yeah totally in agreement here just thought this would be a good jumpoff point for a discussion on maybe setting `refresh: wait_for` as the default for rule execution.

Reply:
The signals index refreshes every 5 seconds automatically, so a quick user might be able to activate a rule, see the success and then check the alerts page before the alerts are available but as soon as they refresh the page again they'll most likely see the alerts.

`refresh: wait_for` as the default could be problematic in some edge cases if lots of rules/tasks are executing, or a user changes the refresh interval to be longer. The request will block until the index refreshes, so the kibana task will take up a worker slot while not doing any actual work -> this hurts the max throughput for tasks. It's worth considering, but since there are tradeoffs to think through so I think the best immediate solution is to be aware of the snafu there and manually force a refresh when necessary, e.g. when we want to check that signals were *not* generated.